src/resolve/remote.rs

```diff
@@ -80,56 +93,64 @@ impl Remote {
         })
     }
 
-    fn content_type(&self, response: &Response) -> Option<ContentType> {
-        // Get a value content type header if any.
-        let header = response.headers().get(header::CONTENT_TYPE)
+    /// Injectively map urls to paths in the download directory.
+    fn target_path(&self, key: &FileKey) -> PathBuf {
+        let extension = key.path.extension();
+
+        let file_hash = Sha256::new()
+            .chain(key.path.to_str().unwrap())
+            .chain(key.mime.as_ref().map(Mime::as_ref).unwrap_or(""))
+            .result();
+
+        let mut hash_name = String::new();
+        file_hash.into_iter().for_each(|x| {
+            hash_name.push(std::char::from_digit((x >> 4).into(), 16).unwrap());
+            hash_name.push(std::char::from_digit((x & 15).into(), 16).unwrap());
+        });
+
+        if let Some(extension) = extension {
+            hash_name.push('.');
+            hash_name.push_str(extension.to_str().unwrap());
+        }
+
+        let mut target = self.temp.clone();
+        target.push(&key.host);
```

Comment:
Why don't you just include the target in the hash? Why do you create a new folder for each domain? I guess that could make debugging better as you don't have to open every single file to find out the one which seems buggy.

Reply:
Pretty much for that reason. I personally wouldn't like it to be a complete blackbox. It's debatable but a bit of a bikeshedding issue.